File: app/src/main/java/org/tianjyan/pwd/activity/GenPasswordDialog.java
package org.tianjyan.pwd.activity;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.Toast;

import org.tianjyan.pwd.R;
import org.tianjyan.pwd.application.PwdGen;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class GenPasswordDialog extends Dialog {

    //region field
    private static final int LENGTH_8 = 8;
    private static final int LENGTH_16 = 16;
    @BindView(R.id.gen_password_lows) CheckBox lows;
    @BindView(R.id.gen_password_caps) CheckBox caps;
    @BindView(R.id.gen_password_numbers) CheckBox numbers;
    @BindView(R.id.gen_password_special) CheckBox special;
    @BindView(R.id.gen_password_len_eight) RadioButton eight;
    @BindView(R.id.gen_password_result) EditText result;
    String genPassword;
    //endregion

    //region onclick
    @OnClick(R.id.gen_password_cancel)
    public void cancelClick() {
        dismiss();
    }

    @OnClick(R.id.gen_password_sure)
    public void sureClick() {
        genPassword = result.getText().toString();
        dismiss();
    }

    @OnClick(R.id.gen_password_gen)
    public void genClick() {
        if(lows.isChecked() || caps.isChecked() || numbers.isChecked() || special.isChecked()) {

            String password = PwdGen.generatePassword(
                    eight.isChecked() ? LENGTH_8 : LENGTH_16,
                    lows.isChecked() ? PwdGen.Optionality.MANDATORY : PwdGen.Optionality.PROHIBITED,
                    caps.isChecked() ? PwdGen.Optionality.MANDATORY : PwdGen.Optionality.PROHIBITED,
                    numbers.isChecked() ? PwdGen.Optionality.MANDATORY : PwdGen.Optionality.PROHIBITED,
                    special.isChecked() ? PwdGen.Optionality.MANDATORY : PwdGen.Optionality.PROHIBITED);
            result.setText(password);
        }
        else {
            Toast.makeText(getContext(), getContext().getString(R.string.gen_password_msg),
                    Toast.LENGTH_SHORT).show();
        }
    }
    //endregion

    //region constructor
    public GenPasswordDialog(Context context) {
        super(context);
    }
    //endregion

    //region function

    //region override
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_gen_password);
        ButterKnife.bind(this);
    }
    //endregion

    //region public
    public String getPassword(){
        return genPassword;
    }

    //endregion

    //endregion
}

File: app/src/main/kotlin/org/tianjyan/pwd/activity/GenPasswordDialog.kt
package org.tianjyan.pwd.activity

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.CheckBox
import android.widget.EditText
import android.widget.RadioButton
import android.widget.Toast
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.home.young.myPassword.R
import org.tianjyan.pwd.application.PwdGen

class GenPasswordDialog(context: Context) : Dialog(context) {
    @BindView(R.id.gen_password_lows) internal var lows: CheckBox? = null
    @BindView(R.id.gen_password_caps) internal var caps: CheckBox? = null
    @BindView(R.id.gen_password_numbers) internal var numbers: CheckBox? = null
    @BindView(R.id.gen_password_special) internal var special: CheckBox? = null
    @BindView(R.id.gen_password_len_eight) internal var eight: RadioButton? = null
    @BindView(R.id.gen_password_result) internal var result: EditText? = null

    lateinit var password: String
        internal set

    @OnClick(R.id.gen_password_cancel)
    fun cancelClick() {
        dismiss()
    }

    @OnClick(R.id.gen_password_sure)
    fun sureClick() {
        password = result!!.text.toString()
        dismiss()
    }

    @OnClick(R.id.gen_password_gen)
    fun genClick() {
        if (lows!!.isChecked || caps!!.isChecked || numbers!!.isChecked || special!!.isChecked) {

            val password = PwdGen.generatePassword(
                    if (eight!!.isChecked) LENGTH_8 else LENGTH_16,
                    if (lows!!.isChecked) PwdGen.Optionality.MANDATORY else PwdGen.Optionality.PROHIBITED,
                    if (caps!!.isChecked) PwdGen.Optionality.MANDATORY else PwdGen.Optionality.PROHIBITED,
                    if (numbers!!.isChecked) PwdGen.Optionality.MANDATORY else PwdGen.Optionality.PROHIBITED,
                    if (special!!.isChecked) PwdGen.Optionality.MANDATORY else PwdGen.Optionality.PROHIBITED)
            result!!.setText(password)
        } else {
            Toast.makeText(context, context.getString(R.string.gen_password_msg),
                    Toast.LENGTH_SHORT).show()
        }
    }

    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_gen_password)
        ButterKnife.bind(this)
    }

    companion object {
        private val LENGTH_8 = 8
        private val LENGTH_16 = 16
    }
}
